### ISSUE TITLE
Fix needed in the credible set clumping function.

### ISSUE THREAD

**Issue by DSuveges** [opentargets/issues#2845]
The credible set clumping [function](https://github.com/opentargets/genetics_etl_python/blob/e0f02ea7237d1f047ae1d7b019cf854d19889825/src/etl/gwas_ingest/clumping.py#L50) needs to be updated. The current implementation works on small datasets and produces the expected output, it fails on the full dataset without meaningful error messages.

**Failed jobs:**

- job [323461f1b3484bcc9c011cc99f1694c6](https://console.cloud.google.com/dataproc/jobs/323461f1b3484bcc9c011cc99f1694c6?region=europe-west1&project=open-targets-genetics-dev)
- job [a512f4e35a2e4b18bd6ff3f67fb0ec21](https://console.cloud.google.com/dataproc/jobs/a512f4e35a2e4b18bd6ff3f67fb0ec21?region=europe-west1&project=open-targets-genetics-dev)
- job [4f3cf1ef83954885978e03337a485567](https://console.cloud.google.com/dataproc/jobs/4f3cf1ef83954885978e03337a485567?region=europe-west1&project=open-targets-genetics-dev)

**Output:**

```
ERROR: (gcloud.dataproc.jobs.wait) Job [a512f4e35a2e4b18bd6ff3f67fb0ec21] failed with error:
Google Cloud Dataproc Agent reports job failure. If logs are available, they can be found at:
https://console.cloud.google.com/dataproc/jobs/a512f4e35a2e4b18bd6ff3f67fb0ec21?project=open-targets-genetics-dev&region=europe-west1
gcloud dataproc jobs wait 'a512f4e35a2e4b18bd6ff3f67fb0ec21' --region 'europe-west1' --project 'open-targets-genetics-dev'
https://console.cloud.google.com/storage/browser/dataproc-staging-europe-west1-234703259993-hrobeqyg/google-cloud-dataproc-metainfo/a71436c5-f4e7-4be6-8446-b86341b35faf/jobs/a512f4e35a2e4b18bd6ff3f67fb0ec21/
gs://dataproc-staging-europe-west1-234703259993-hrobeqyg/google-cloud-dataproc-metainfo/a71436c5-f4e7-4be6-8446-b86341b35faf/jobs/a512f4e35a2e4b18bd6ff3f67fb0ec21/driveroutput
```

So not much to work on with... I have a few ideas to start.

**Comment by DSuveges** [comment]
That function has this row:
```Python
# Counting the number of occurrence for each pair - for one study, each pair should occure only once:
.withColumn("rank", f.count(f.col("tagVariantId")).over(w))
```
Where the applied window looks like this:
```Python
w = Window.partitionBy("studyId", "variantPair").orderBy(f.col("negLogPVal").desc())
```

This introduces a problem: it creates circular dependency for leads with the same p-values. It needs to be resolved by:
```Python
.withColumn("rank", f.row_number().over(w))
```

**Comment by DSuveges** [comment]
This update would not solve the issue, the full dataset still not run. What we try next is to represent referencing  leads as small graphs. Then we try to resolve the root of each graph.

**Comment by DSuveges** [comment]
This effort is closed by the [PR#64](https://github.com/opentargets/genetics_etl_python/pull/64) in the ETL repo.

**Comment by DSuveges** [comment]
This effort is closed by the [PR#64](https://github.com/opentargets/genetics_etl_python/pull/64) in the ETL repo.
